### PR TITLE
Frontend for reset page

### PR DESCRIPTION
**Pull request by burtonjong**
Didnt test it yet but the input looks good to me

didnt use react hook form yet either but that can be done later

only problem maybe is this:

```
{
    "scoreComponents": "[\n              {\n                id: \"Cool beans, this should be a uuid\",\n                friendlyName: \"no name for you\",\n                isSidepot: false\n              },\n              {\n                id: \"uuid v2\",\n                friendlyName: \"I love eating free food\",\n                isSidepot: false\n              }\n            ]",
    "scoringSidepots": "[\n              {\n                id: \"Cool beans, this should be a uuid\",\n                friendlyName: \"no name for you\",\n                isSidepot: false\n              },\n              {\n                id: \"uuid v2\",\n                friendlyName: \"I love eating free food\",\n                isSidepot: false\n              }\n            ]",
    "startDate": "2024-08-02",
    "endDate": "2024-08-09",
    "resetUsers": true,
    "resetRooms": true,
    "resetTeams": true
}
```

It copies the \n

### CODE REVIEW

**Review by ideen1 — COMMENTED** [review]
@burtonjong Let me know once the JSON fields are also done

**Comment by burtonjong** [comment]
@ideen1 hopefully tomorrow just got back from camping today

**Comment by burtonjong** [comment]
Again its not tested but technically it should work - also note that i had to rename some of the fields in resource.ts for readability. I also had to create a special Type so that scoringComponents and scoringSidepots are considered arrays (or else there were type errors)

**Comment by burtonjong** [comment]
image


Here is the object that is passed to the mutation for reference

**Comment by burtonjong** [comment]
image

And how it looks

File: src/components/reset/ResetPage.tsx [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+        }
+        console.log("Form Data Submitted:", input, statusCode?.statusCode);
+        return;
+      } catch (error) {
+        console.error("Error updating user", error);
+        throw error;
+      }
+    },
+  });
+  type Arguments = Pick<
```
Comment (ideen1) [inline src/components/reset/ResetPage.tsx:40]: Everything look great. Can you just elaborate on why this type was needed?

File: src/components/reset/ResetPage.tsx [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+import { Button, CheckboxField, Input, Label } from "@aws-amplify/ui-react";
+import { useMutation } from "@tanstack/react-query";
+
+const client = generateClient<Schema>();
+export default function ResetPage() {
+  const userMutation = useMutation({
+    mutationFn: async (input: Schema["ResetHackathon"]["args"]) => {
+      console.log(input);
+      try {
+        // e.preventDefault(); // This will prevent the default form submit action which is to refresh the page
```
Comment (ideen1) [inline src/components/reset/ResetPage.tsx:16]: Remove commented code

File: src/components/reset/ResetPage.tsx [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+  }
+  return (
+    <form
+      onSubmit={handleSubmit(onSubmit)}
+      className="relative flex w-full flex-col justify-center gap-4 bg-white p-4 md:p-8"
+    >
+      <div className="flex w-full flex-col justify-between gap-2 md:gap-12">
+        <div className="flex w-full flex-row">
+          <div className="mr-24 flex w-1/3 flex-col gap-2">
+            <Label>Score Components (JSON): </Label>
```
Comment (ideen1) [inline src/components/reset/ResetPage.tsx:116]: Can you remove (JSON)

File: src/components/reset/ResetPage.tsx [hunk 4]
```diff
@@ -0,0 +1,251 @@
...
+                  isSidepot: false,
+                  id: generateId(),
+                })
+              }
+            >
+              Add Score Component
+            </Button>
+          </div>
+          <div className="flex w-1/3 flex-col gap-2">
+            <Label htmlFor="scoringSidepots">Scoring Sidepots (JSON): </Label>
```
Comment (ideen1) [inline src/components/reset/ResetPage.tsx:149]: Can you remove (JSON)

File: src/components/reset/ResetPage.tsx [hunk 5]
```diff
@@ -0,0 +1,251 @@
...
+            <CheckboxField label="Reset Rooms: " {...register("resetRooms")} />
+            <CheckboxField
+              label="Reset Scores: "
+              {...register("resetScores")}
+            />
+          </div>
+        </div>
+
+        <div className="flex flex-col gap-2">
+          <Label htmlFor="safetyCheck">Safety Check: </Label>
```
Comment (ideen1) [inline src/components/reset/ResetPage.tsx:219]: Rename Label to: **Enter "deletehackathon" to confirm**

```suggestion
          <Label htmlFor="safetyCheck">Enter "deletehackathon" to confirm </Label>
```

File: src/components/reset/ResetPage.tsx [hunk 6]
```diff
@@ -0,0 +1,251 @@
...
+          <div className="flex w-1/2 flex-row gap-2">
+            <Input
+              required
```
Comment (ideen1) [inline src/components/reset/ResetPage.tsx:222]: Add placeholder that says "deletehackathon"

**Review by ideen1 — CHANGES_REQUESTED** [review]

**Comment by burtonjong** [comment]
@ideen1 in handler.ts (allan's function) it is "delete hackathon" (with a space), so that is what i put

**Review by ideen1 — APPROVED** [review]
LGTM!!

**Pull request merged**